client/common.go

```diff
@@ -165,7 +165,11 @@ func setupRancherBaseClient(rancherClient *RancherBaseClientImpl, opts *ClientOp
 		return err
 	}
 
-	defer resp.Body.Close()
+	defer func(body io.Closer) {
+		if closeErr := body.Close(); closeErr != nil && debug {
+			fmt.Printf("Warning: failed to close schema response body: %v\n", closeErr)
```

Comment:
```suggestion
			logrus.WithError(closeErr).Warn("Failed to close schema response body")
```

Reply:
Apply to other changes.

Reply:
Agree to use logrus instead, but, do we have any concern remove debug flag and change to logrus in the rest part of client/common.go?